txpool/operator.go

```diff
@@ -50,7 +50,30 @@ func (p *TxPool) AddTxn(ctx context.Context, raw *proto.AddTxnReq) (*proto.AddTx
 }
 
 // Subscribe implements the operator endpoint. It subscribes to new events in the tx pool
-func (p *TxPool) Subscribe(req *empty.Empty, stream proto.TxnPoolOperator_SubscribeServer) error {
-	// TODO
-	return nil
+func (p *TxPool) Subscribe(
+	request *proto.SubscribeRequest,
+	stream proto.TxnPoolOperator_SubscribeServer,
+) error {
+	subscription := p.eventManager.subscribe(request.Types)
+
+	teardown := func() error {
```

Comment:
Do you need to return error now? If you don't return error, you can use `defer`

Reply:
Good catch 💯 

This is leftover from an earlier iteration of the implementation.

Fixed 👍 

167e38f